pages/demo/form.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <div class="p-4 space-y-4">
+    <h2 class="text-lg font-bold">Demo: Form</h2>
+    <TForm
+      v-model="data"
+      :fields="fields"
+      :field-config="{ labelPosition: 'top' }"
+      class="space-y-4"
+      submit-label="Submit"
+      @save="submit"
+    />
+  </div>
+</template>
+
+<script>
+import {
+  required,
+  email,
+  minLength,
+  alphaNum,
+  helpers
+} from '@vuelidate/validators'
+import { ref } from '@nuxtjs/composition-api'
+import { useDoc } from '~/use/doc'
+export default {
+  name: 'Form',
+  setup() {
+    const { find, exists } = useDoc('profiles')
+    const data = ref({
+      name: '',
+      email: '',
+      username: '',
+      place: ''
+    })
+    const isUnique = async (value) => {
+      await find('username', value)
+      if (exists.value) {
+        return false
+      } else {
+        return true
+      }
+    }
+
+    const ErrorMessages = {
+      name: {
+        required: 'validation.name.required',
+        minLength: 'validation.name.minLength'
+      },
+      email: {
+        required: 'validation.email.required',
+        email: 'validation.email.email'
+      },
+      username: {
+        required: 'validation.username.required',
+        minLength: 'validation.username.minLength',
+        isUnique: 'validation.username.isUnique',
+        alphaNum: 'validation.username.alphaNum'
+      },
+      place: {
+        required: 'validation.place.required'
+      }
+    }
+    const fields = [
+      {
+        name: 'name',
+        key: 'account.name',
+        validations: {
+          required: helpers.withMessage(ErrorMessages.name.required, required),
```

Comment:
Can you please simplify config for validation somehow? Something similar to how I did it with types. So that you can just specify "required: true" or ['required', 'minLength'].